lib/SIL/Linker.cpp

```diff
@@ -157,10 +157,9 @@ void SILLinkerVisitor::linkInVTable(ClassDecl *D) {
 //                                  Visitors
 //===----------------------------------------------------------------------===//
 
-static template<typename Inst>
-bool applyInstCalleeIsGeneric(Inst AI) {
-  return AI->getCallee()->getType().template castTo<SILFunctionType>()
-           ->getGenericSignature();
+template<typename Inst>
+static bool applyInstCalleeIsGeneric(Inst AI) {
```

Comment:
You can just remove this method, and remove the 'if' statements that use it. Calling visitApplySubstitutions() unconditionally is OK.

Reply:
@slavapestov Thanks, I've reset the pull request.